jcl/src/openj9.traceformat/share/classes/com/ibm/jvm/traceformat/TraceFormat.java

```diff
@@ -355,21 +356,23 @@ public static void main(String[] args) throws Exception {
 				thread = current;
 				output.println(formatted.toString());
 			}
-			
+
 			/* print percentage */
 			if (context.getTotalRecords() != recordsProcessed) {
 				recordsProcessed = context.getTotalRecords();
 				float processedMbytes = ((float) (context.getTotalRecords() * context.getRecordSize())) / (1024 * 1024);
 				if (processedMbytes % 10 == 0) {
-					int percent = (int) ((processedMbytes * 100) / (totalBytes/(1024*1024)));
+					int percent = (int) ((processedMbytes * 100) / (totalBytes / (1024 * 1024)));
 					if (verbose.booleanValue()) {
-						end = System.currentTimeMillis();
-						float MbpsBlock = (float) 10.0 / (float) ((end - startBlock) / 1000.0);
-						float Mbps = (float) (processedMbytes) / (float) (((float) (end - start) / 1000.0));
-						startBlock = System.currentTimeMillis();
-						context.message(context, "Processed " + processedMbytes + "Mb ("+percent+"%), burst speed: "+MbpsBlock+"Mb/s, average: "+Mbps+"Mb/s");
+						end = System.nanoTime();
+						final long SECOND_SCALE = 1000 * 1000 * 1000;
+						float MbpsBlock = (float) 10.0 / (end - startBlock) * SECOND_SCALE;
+						float Mbps = processedMbytes / (end - start) * SECOND_SCALE;
```

Comment:
I think this computes the wrong value (by a factor of 1e6) for `MbpsBlock`. Doing the multiplication before the division avoids some unwanted rounding:
```
	float MbpsBlock = 1024.0f * 1024.0f * SECOND_SCALE / (end - startBlock);
	float Mbps = processedMbytes * SECOND_SCALE / (end - start);
```

Reply:
> I think this computes the wrong value (by a factor of 1e6) for MbpsBlock

What speed does `MbpsBlock` represent, is the existing code already wrong - `float MbpsBlock = (float) 10.0 / (float) ((end - startBlock) / 1000.0)`?

Reply:
I think `MbpsBlock` is intended to be the processing rate of the most recent block, expressed in millions of bytes per second. The block is 10MB (that's where the `10.0` comes in); dividing by 1000 converts milliseconds to seconds: the result has units of MB/s.

When time is expressed in nanoseconds instead of milliseconds, we need to replace 1000 with 1e9. My corrected suggestion is:
```
    float MbpsBlock = 10.0f * SECOND_SCALE / (end - startBlock);
```
This is equivalent to your proposal, improved by multiplying before dividing (in fact, I expect that multiplication to happen at compile time).

Reply:
Updated.